core/src/main/java/com/linecorp/armeria/server/HttpHeaderUtil.java

```diff
@@ -146,57 +147,21 @@ private static InetSocketAddress createInetSocketAddress(String address) throws
 
         // Remote quotes. e.g. "[2001:db8:cafe::17]:4711" => [2001:db8:cafe::17]:4711
         final String addr = firstChar == '"' ? QUOTED_STRING_TRIMMER.trimFrom(address) : address;
-
-        final InetSocketAddress inetSocketAddress;
         try {
-            if (addr.charAt(0) == '[') {
-                inetSocketAddress = createInetSocketAddress(addr, ']');
-            } else {
-                inetSocketAddress = createInetSocketAddress(addr, ':');
+            final HostAndPort hostAndPort = HostAndPort.fromString(addr);
+            final byte[] addressBytes = NetUtil.createByteArrayFromIpAddressString(hostAndPort.getHost());
+            if (addressBytes == null) {
+                logger.debug("Fail to parse an address: {}", address);
+                return null;
             }
-            // Log debug message if it is an invalid address.
-            if (inetSocketAddress == null) {
-                logger.debug("Failed to parse an address: {}", address);
-            }
-            return inetSocketAddress;
-        } catch (NumberFormatException e) {
-            logger.debug("Fail to parse a port from an address: {}", address, e);
+            return new InetSocketAddress(InetAddress.getByAddress(addressBytes),
+                                         hostAndPort.getPortOrDefault(0));
+        } catch (IllegalArgumentException e) {
+            logger.debug(e.getMessage(), e);
```

Comment:
I wouldn't rely on `e.getMessage()` because it may be `null` although `createByteArrayFromIpAddressString()` always throws with a message.

Reply:
`logger.debug("Failed to parse an address: {}, address, e);`